### ISSUE TITLE
Replace the cursor with the fill bucket icon when in fill mode

### ISSUE THREAD

**Issue by fsih** [LLK/scratch-paint#358]
Whenever the fill tool is active and the mouse is over the paint canvas, the usual mouse cursor should disappear, and be replaced with the fill tool icon (located at `scratch-paint/src/components/fill-mode/fill.svg`)

The icon should be lined up with the pointer location like this:
![image](https://user-images.githubusercontent.com/2855464/38269715-43d7cf68-374f-11e8-8007-4c688f323aa3.png)

/cc @carljbowman

**Comment by Jacco** [comment]
I have been thinking about this feature. And there are a few things that might need some discussion:

If the actual cursor is hidden and a moving icon is shown using the suggested svg object on the paper canvas a few issues arise:
  - when at the edges of the paper the icon partly will dissappear
  - the icon has no semi-transparent dropshadow that makes clear that the cursor is not part of the drawing
  - th icon is grey in stead of the usual black
  - the icon has no white surrounding is for contrast enhancing when hovering over dark areas
  - the icon has two tranparent holes that should not be transparent (side and top of bucket)

Most of these issues can be solved by creating a new svg. 

It is for most browsers also possible to set custom cursors (css property cursor with a url value). These will not have the edge problems.

Also I would like to suggest a feature:
- it would be nice if the paint bucket svg would show the color that is going to be applied.

Css code:
```
  cursor: url(page date url here) 25 10, auto;
```

You can create data url here:
[dataurl](http://dataurl.net/)

**Comment by fsih** [comment]
I agree we should use the CSS cursor property. An easy way to get to it is
`paper.view.element.style.cursor =`

I think you're right that we might want to adjust the fill icon a bit to make it work better as a cursor
 - Change the color?
 - Fill the holes in the SVG
 - Add a drop shadow
 - Change the format to something other than SVG for compatibility purposes?

Since this needs some design, I will take off help wanted for now and put it back on when we have that ready.

For the paint bucket showing the color that's going to be applied, I think that's a cool idea! You should file it as a feature request.

/cc @carljbowman

**Comment by fsih** [comment]
Here is the new cursor file we would like to use!
[fill-cursor.zip](https://github.com/LLK/scratch-paint/files/1900618/fill-cursor.zip)
